### ISSUE TITLE
[Feature Request]: Dump audio, models

### ISSUE THREAD

**Issue by mk360** [PCSX2/pcsx2#12105]
### Description

Since we're already able to dump textures & mipmaps, I was wondering if we could also dump audio clips and files in the same vein (writing them to disk when they're first loaded by the game). I can give audio dumping a shot, if someone can point me to the right direction. Would be even nicer if we could "hot-replace" audio clips the same way we can replace textures.

### Reason

Texture dumping is a good first step to create skin mods or green screen backgrounds for video clip edits, but it would be nice if we could extract more assets from the games.

### Examples

The audio / model dump would work similarly to the current texture dump implementation: while the toggle is enabled, whenever the emulator reads an audio file from the iso, it can write it to disk in mp3 format.

**Comment by mk360** [comment]
@Mrlinkwii not exactly, no. It'd work, as I mentioned in my initial post, just like textures: when the emulator loads any texture of any sort, it gets dumped to the disk. Ideally audio dumping would work the same way.

**Comment by Felipe379** [comment]
There's a fork that actually implements a sort of prototype for 3D Screenshot: [feature/3d-screenshot/](https://github.com/scurest/pcsx2)
This would allow you to easily get models, but models will always be stretched along the depth axis, which you can work around by scaling around the model.

However, there seems to exist a tool that fixes the stretched models: [PCSX2ModelConverter](https://www.vg-resource.com/thread-20541-post-514141.html#pid514141)

Luckily the source code has been included along with the tool, so maybe it's possible to implement 3D Screenshots along with whatever PCSX2ModelConverter does to fix the models?

**Comment by TheMysteriousMirageofOld** [comment]
I'm just a random user of PCSX2, but I would definitely be a fan of, at least, Audio dumping becoming a thing. It's pretty clear that many games are loading audio files from the disk and playing them so they have to be entering memory at one point or another. 

Trying to rip a game disk's audio files is a massive pain in the ass if the files are hidden and you aren't good at getting rid of copy protection or finding a way to unhide the files, since it loads in the whole ass audio in a long single file on a good day. And often it's not even played at the correct speed. In PCSX2, when the game is loaded, this is often not an issue (especially nowadays), so clearly the software is already doing something to load identify the files and load them to play and play nothing else.

I think the idea would be that it would dynamically splice the audio that's loaded so the headache inducing route stops being the only proper way. Texture dumping has similarly helped me when I've looked at some obscure games that happen to have no files visible in the .ISO despite containing a lot of them no doubt.

I also think a lot of people who are fans of video games would love if they could find a way to dump audio from even the most obscure old titles they have, too.
